File: CPP/Implementation2_thread_unsafe/triangle.cpp
#include "triangle.h"

triangle::triangle()
{}

triangle::triangle(const triangle& t)
{
	p = t.p;
}

position triangle::get_position()
{
	return p;
}
triangle* triangle::set_position(const position& pos)
{
	p =pos;
	return this;
}
void triangle::move_to(position where)
{
	ms.move_to(this,where);
}
void triangle::move_to(int dx, int dy)
{

	ms.move_to(this,dx,dy);
}
void triangle::move_by(int dx, int dy)
{
	ms.move_by(this,dx,dy);
}	
void triangle::move_by(position where)
{	
	ms.move_by(this,where);
}

File: JAVA/Implementation2_thread_safe/Triangle.java
public class Triangle implements Movable {
	
	private Position p;
	private MoveServant m;
	public Triangle()
	{
		m = new MoveServant();
	}
        
	public void setPosition(Position p) {
		this.p = p;
	}

	public Position getPosition() {
		return this.p;
	}
	
	public void moveTo(int dx, int dy)
	{
		m.moveTo(this,dx,dy);
	}
	public void moveTo(Position p)
	{
		m.moveTo(this,p);
	}
	public void moveBy(Position where)
	{
		m.moveBy(this,where);
	}
	public void moveBy(int dx, int dy)
	{
		m.moveBy(this,dx,dy);
	}

}